File: 2GII/PDOO/QytetetJava/src/modeloqytetet/Sorpresa.java
package modeloqytetet;

public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    // ------------------------- Constructor -----------------------------------
    public Sorpresa(String texto, int valor, TipoSorpresa tipo) {
        this.texto = texto;
        this.valor = valor;
        this.tipo = tipo;
    }
    
    // ------------------------- Consultores -----------------------------------   
    String getTexto() {
        return texto;
    }
    
    TipoSorpresa getTipo() {
        return tipo;
    }
    
    int getValor() {
        return valor;
    }
    
    // --------------------------- Métodos -------------------------------------
    @Override
    public String toString() {
        return "~ Texto: " + texto + "\n~ Valor: " + valor + "\n~ Tipo: " + tipo;
    }
}

File: 2GII/PDOO/QytetetRuby/lib/sorpresa.rb
#encoding: utf-8

require_relative "tipo_sorpresa"

module ModeloQytetet

  class Sorpresa
    attr_reader :texto, :tipo, :valor

    # -------------------------- Constructor -----------------------------------    
    # @pre Texto - string, valor - int, tipo - TipoSorpresa
    def initialize(texto, valor, tipo)
      @texto = texto
      @valor = valor
      @tipo = tipo
    end

    # ---------------------------- Métodos -------------------------------------
    def to_s
      "\n~ Texto: #{@texto} \n~ Valor: #{@valor} \n~ Tipo: #{@tipo}"
    end

  end
  
end